### PR TITLE
Make implicit fields coming from source query expressions use string id

### LINKED ISSUE #28451: Nested questions get confused around custom columns that are just columns
Gist: make a card that selects a custom column which is just another column. Save the question. Make a question based on this question. Error.

Fleshed out:
New mbql question: Sample Dataset > Orders. Select just Total. Add a custom column who's definition is just `[Tax]` and a limit of 2. (selecting few columns and limit of 2 is not required, just makes viewing metadata waaay easier)
```clojure
{:source-table 2,
 :fields [[:field 40 nil] [:expression "custom"]],
 :expressions {"custom" [:field 37 nil]},
 :limit 2}
```
Make a new question based on this saved question.
image
image

> Column "source.TAX" not found; SQL statement:

generated sql:

```sql
SELECT "source"."TOTAL" AS "TOTAL",
       "source"."TAX" AS "TAX" -- selecting TAX not custom
FROM (SELECT "source"."TOTAL" AS "TOTAL",
             "source"."custom" AS "custom" -- no TAX here
      FROM (SELECT "PUBLIC"."ORDERS"."ID" AS "ID",
                   "PUBLIC"."ORDERS"."USER_ID" AS "USER_ID",
                   "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID",
                   "PUBLIC"."ORDERS"."SUBTOTAL" AS "SUBTOTAL",
                   "PUBLIC"."ORDERS"."TAX" AS "TAX",
                   "PUBLIC"."ORDERS"."TOTAL" AS "TOTAL",
                   "PUBLIC"."ORDERS"."DISCOUNT" AS "DISCOUNT",
                   "PUBLIC"."ORDERS"."CREATED_AT" AS "CREATED_AT",
                   "PUBLIC"."ORDERS"."QUANTITY" AS "QUANTITY",
                   "PUBLIC"."ORDERS"."TAX" AS "custom"
            FROM "PUBLIC"."ORDERS")
      "source" LIMIT 2)
"source" LIMIT 1048575
```

I think the issue is that the custom-column has an id on it and the query processor ignores the analysis in favor of "oh, an id. i know this field. it is TAX".

```clojure
  (db/select-one-field :result_metadata 'Card :id 4)
  ({:name "TOTAL",
    :settings nil,
    :field_ref [:field 40 nil],
    :effective_type :type/Float,
    :id 40,
    :display_name "Total",
    :base_type :type/Float}
   {:name "custom",
    :settings nil,
    :field_ref [:expression "custom"],
    :effective_type :type/Float,
    :id 37,  ;; reaches out to "TAX" instead of "custom"
    :display_name "custom",
    :base_type :type/Float})
```

### PR DESCRIPTION
**Pull request by lbrdnk**
Closes #28451

### Description

When implicit fields are generated for a source query and that query contains expression that is just field, field in question is generated twice and later result is discarded. To overcome this, field generated from expression is in string id form.

### How to verify

Run the new test or try reproduction steps from the issue in question.

### Checklist

- [X] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by metamben — APPROVED** [review]
LGTM

File: test/metabase/query_processor/middleware/add_implicit_clauses_test.clj [hunk 1]
```diff
@@ -152,6 +152,18 @@
                          {:source-query    source-query
                           :source-metadata source-metadata}))))))))
 
+(deftest ^:parrallel expression-with-only-field-in-source-query-test
```
Comment (nemanjaglumac) [inline test/metabase/query_processor/middleware/add_implicit_clauses_test.clj:155]: I was comparing diffs between this and the backport. This seems to be a typo.

```diff
- (deftest ^:parrallel expression-with-only-field-in-source-query-test
+ (deftest ^:parallel expression-with-only-field-in-source-query-test
```

Comment (lbrdnk) [reply test/metabase/query_processor/middleware/add_implicit_clauses_test.clj:155]: Yes. Found that while backporting. Will correct in master shortly.

Comment (metamben) [reply test/metabase/query_processor/middleware/add_implicit_clauses_test.clj:155]: The linter should have complained.

**Pull request merged**